tests/TestCase/Validation/ValidationTest.php

```diff
@@ -2529,6 +2529,103 @@ public function testCompareWith()
+        $this->assertFalse(Validation::containsNonAlphaNumeric('ABCDEFGHIJKLMNOPQRSTUVWXYZ'));
+        $this->assertFalse(Validation::containsNonAlphaNumeric('0123456789'));
+        $this->assertFalse(Validation::containsNonAlphaNumeric('abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789'));
+
+        $this->assertTrue(Validation::containsNonAlphaNumeric('#'));
+        $this->assertTrue(Validation::containsNonAlphaNumeric("0\n"));
+        $this->assertTrue(Validation::containsNonAlphaNumeric("\n"));
+        $this->assertTrue(Validation::containsNonAlphaNumeric("\t"));
+        $this->assertTrue(Validation::containsNonAlphaNumeric("\r"));
+        $this->assertTrue(Validation::containsNonAlphaNumeric(" "));
+
+        $this->assertTrue(Validation::containsNonAlphaNumeric('#abcdef'));
+        $this->assertTrue(Validation::containsNonAlphaNumeric('abc#def'));
+        $this->assertTrue(Validation::containsNonAlphaNumeric('abcdef#'));
+        $this->assertTrue(Validation::containsNonAlphaNumeric('abc def'));
+        $this->assertTrue(Validation::containsNonAlphaNumeric("abcdef\n"));
+
+        $this->assertTrue(Validation::containsNonAlphaNumeric('##abcdef', 2));
+        $this->assertTrue(Validation::containsNonAlphaNumeric('abcdef##', 2));
+        $this->assertTrue(Validation::containsNonAlphaNumeric('#abcdef#', 2));
+        $this->assertTrue(Validation::containsNonAlphaNumeric('#abc#def', 2));
+        $this->assertTrue(Validation::containsNonAlphaNumeric('abc#def#', 2));
+
+        $this->assertTrue(Validation::containsNonAlphaNumeric('#♥abcdef', 2));
+        $this->assertTrue(Validation::containsNonAlphaNumeric('abcdef#♥', 2));
+        $this->assertTrue(Validation::containsNonAlphaNumeric('#abcdef♥', 2));
+        $this->assertTrue(Validation::containsNonAlphaNumeric('#abc♥def', 2));
+        $this->assertTrue(Validation::containsNonAlphaNumeric('abc#def♥', 2));
+
+        $this->assertTrue(Validation::containsNonAlphaNumeric('###abcdef', 2));
+        $this->assertTrue(Validation::containsNonAlphaNumeric('abc###def', 2));
+        $this->assertTrue(Validation::containsNonAlphaNumeric('abcdef###', 2));
+        $this->assertTrue(Validation::containsNonAlphaNumeric('#abc#def#', 2));
+
+        $this->assertFalse(Validation::containsNonAlphaNumeric('##abcdef', 3));
+        $this->assertFalse(Validation::containsNonAlphaNumeric('abcdef##', 3));
+        $this->assertFalse(Validation::containsNonAlphaNumeric('abc##def', 3));
+        $this->assertFalse(Validation::containsNonAlphaNumeric('ab#cd#ef', 3));
+    }
+
+    /**
+     * testContainsNonAlphaNumeric method with array
+     *
+     * @return void
+     */
+    public function testContainNonAlphaNumericPassedAsArray()
+    {
+        $this->assertFalse(Validation::containsNonAlphaNumeric(['badCheckKey' => 'abcdef']));
+        $this->assertFalse(Validation::containsNonAlphaNumeric(['check' => '0']));
```

Comment:
I don't think you need to handle the array case at all. Its not required for validators any more.